tests/measurements/test_measurements.py

```diff
@@ -206,19 +219,23 @@ def circuit():
 class TestProperties:
     """Test for the properties"""
 
+    def test_return_type(self):
+        """Test MeasurementProcess ``return_type`` property is None."""
+        assert MeasurementProcess().return_type is None
```

Comment:
Because it has no abstract methods. Maybe since I removed the `return_type` abstract property it is not an abstract class anymore...

I moved the inheritance from `ABC` to its child classes `SampleMeasurement`, `StateMeasurement` and `CustomMeasurement`.

Let me know what you think!

Reply:
Moving this inheritance we let users instantiate a `MeasurementProcess` class, but if they execute it won't do anything

Reply:
Actually, can we mark `MeasurementProcess` as an `ABC` anyways? That way we can communicate that it's not "meant" to be instantiated, even though it could technically be.

Reply:
Done!